providers/test/Test.fst

```diff
@@ -359,126 +374,347 @@ let test_dh () : St unit =
+    features_aesni = aesni;
+    features_shaext = shaext;
+  }
+
+inline_for_extraction
+let f_none : features = {
+  features_avx = false;
+  features_avx2 = false;
+  features_bmi2 = false;
+  features_adx = false;
+  features_aesni = false;
+  features_shaext = false;
+}
+
+inline_for_extraction
+let f_avx : features = // [@inline_let] ({ f_none with features_avx = true; })
+{
+  features_avx = true;
+  features_avx2 = false;
+  features_bmi2 = false;
+  features_adx = false;
+  features_aesni = false;
+  features_shaext = false;
+}
+
+inline_for_extraction
+let f_avx2 : features = // [@inline_let] ({ f_none with features_avx2 = true; })
+{
+  features_avx = false;
+  features_avx2 = true;
+  features_bmi2 = false;
+  features_adx = false;
+  features_aesni = false;
+  features_shaext = false;
+}
+
+inline_for_extraction
+let f_bmi2 : features = // [@inline_let] ({ f_none with features_bmi2 = true; })
+{
+  features_avx = false;
+  features_avx2 = false;
+  features_bmi2 = true;
+  features_adx = false;
+  features_aesni = false;
+  features_shaext = false;
+}
+
+inline_for_extraction
+let f_adx : features = // [@inline_let] ({ f_none with features_adx = true; })
+{
+  features_avx = false;
+  features_avx2 = false;
+  features_bmi2 = false;
+  features_adx = true;
+  features_aesni = false;
+  features_shaext = false;
+}
+
+inline_for_extraction
+let f_aesni : features = // [@inline_let] ({ f_none with features_aesni = true; })
+{
+  features_avx = false;
+  features_avx2 = false;
+  features_bmi2 = false;
+  features_adx = false;
+  features_aesni = true;
+  features_shaext = false;
+}
+
+inline_for_extraction
+let f_shaext : features = // [@inline_let] ({ f_none with features_shaext = true; })
+{
+  features_avx = false;
+  features_avx2 = false;
+  features_bmi2 = false;
+  features_adx = false;
+  features_aesni = false;
+  features_shaext = true;
+}
+
+inline_for_extraction
+type impl = | Hacl | Vale | OpenSSL | BCrypt
+
+inline_for_extraction
+let config = (impl & features)
+
+inline_for_extraction
+let check_static_config (c: config) : Stack bool (fun _ -> True) (fun _ _ _ -> True) =
+  match c with
+  | (i, f) ->
+    AC.init ();
+    let no_avx = not (AC.has_avx ()) in
+    let no_avx2 = not (AC.has_avx2 ()) in
+    let no_bmi2 = not (AC.has_bmi2 ()) in
+    let no_adx = not (AC.has_adx ()) in
+    let no_aesni = not (AC.has_aesni ()) in
+    let no_shaext = not (AC.has_shaext ()) in
+    if
+      (f.features_avx && no_avx) ||
+      (f.features_avx2 && no_avx2) ||
+      (f.features_bmi2 && no_bmi2) ||
+      (f.features_adx && no_adx) ||
+      (f.features_aesni && no_aesni) ||
+      (f.features_shaext && no_shaext)
+    then
+      false
+    else
+      match i with
+      | Hacl -> SC.hacl
+      | Vale -> SC.vale
+      | OpenSSL -> SC.openssl
+      | BCrypt -> SC.bcrypt
+
+#push-options "--z3rlimit 16"
+
+inline_for_extraction
+let set_config (c: config) : Stack unit (fun _ -> True) (fun _ _ _ -> True) =
```

Comment:
so there's an understanding here that the client always calls `check_config` before calling `set_config`... can there be a comment to that effect?